### PR TITLE
Feature upsert template

### PR DESCRIPTION
**Pull request by pepcitron**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 
+## [1.1.5] - 
+### Added
+- `updateTemplate` - update a template for a given module working copy
+- `upsertTemplate` - update (create or update) a template for a given module working copy
```
Comment (Lucas-C) [inline CHANGELOG.md]: Typo: update -> upsert

File: src/com/vsct/dt/hesperides/jenkins/pipelines/Hesperides.groovy [hunk 2]
```diff
@@ -261,6 +261,43 @@ class Hesperides implements Serializable {
...
+
+    def upsertTemplate(Map args) { required(args, ['moduleName', 'moduleVersion', 'location', 'filename', 'content']) // optional: title
+        try {
+            args.version_id = getTemplate(args).version_id
+            updateTemplate(args)
+        } catch (HttpException httpException) {
+            if (httpException.statusCode != 404) {
+                throw ex
+            }
+            log("------------ template name for this module not found")
```
Comment (Lucas-C) [inline src/com/vsct/dt/hesperides/jenkins/pipelines/Hesperides.groovy]: Debug log ?

File: vars/hesperides.txt [hunk 3]
```diff
@@ -148,4 +148,15 @@
         updateProperties app: 'CSC', platform: 'USN1', jsonPropertyUpdates: "$env.WORKSPACE/hesperides_changes/CSC-USN1.json", commitMsg: 'Test'
         </code></pre>
     </section>
+
+    <section>
+        <h1>hesperides.upsertTemplate</h1>
+        Upsert (create or update) a template on a given module. Existing template for a version module is replaced.
```
Comment (Lucas-C) [inline vars/hesperides.txt]: Typo: "version module" -> "module version"

File: CHANGELOG.md (hunk 1 above)
Comment (pepcitron) [reply CHANGELOG.md]: Done

File: src/com/vsct/dt/hesperides/jenkins/pipelines/Hesperides.groovy (hunk 2 above)
Comment (pepcitron) [reply src/com/vsct/dt/hesperides/jenkins/pipelines/Hesperides.groovy]: We remove the line (because http requests are enough)

**Comment by Lucas-C** [comment]
Note : TravisCI is failing because of some trailing whitespaces in `CHANGELOG.md`

You can install the pre-commit hooks locally to detect this at commit time

File: vars/hesperides.txt (hunk 3 above)
Comment (pepcitron) [reply vars/hesperides.txt]: Hawk eye :)

**Comment by pepcitron** [comment]
Review feedback done, squash too

**Comment by Lucas-C** [comment]
Thank you !

**Pull request merged**